cmd/jobobject-util/main.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/Microsoft/hcsshim/internal/jobobject"
+	"github.com/Microsoft/hcsshim/internal/processorinfo"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	processorLimitFlag  = "processorLimit"
+	processorCountFlag  = "processorCount"
+	processorWeightFlag = "processorWeight"
+	memoryLimitInMBFlag = "memoryLimitInMB"
+	affinityFlag        = "affinity"
+
+	usage = `jobobject-util is a command line tool for getting and setting job object limits`
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "jobobject-util"
+	app.Commands = []cli.Command{
+		getJobObjectLimitsCommand,
+		setJobObjectLimitsCommand,
+	}
+	app.Usage = usage
+
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+var getJobObjectLimitsCommand = cli.Command{
+	Name:      "get",
+	Usage:     "gets the container's resource limits",
+	ArgsUsage: "get [flags] <container-id>",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  processorLimitFlag,
+			Usage: "Optional: get container's processer limit",
+		},
+		cli.BoolFlag{
+			Name:  processorCountFlag,
+			Usage: "Optional: get container's processor count",
+		},
+		cli.BoolFlag{
+			Name: processorWeightFlag,
+			Usage: `Optional: get container's processor weight.
+			Note: Due to rounding issues as a result of the job object's cpu weight being represented as an integer, 
+			the resulting processor weight may not correspond to the same processor weight initially set.`,
+		},
+		cli.BoolFlag{
+			Name:  memoryLimitInMBFlag,
+			Usage: "Optional: get container's memory limit in MBs.",
+		},
+		cli.BoolFlag{
+			Name:  affinityFlag,
+			Usage: "Optional: get container's processor affinity as a bitmask.",
+		},
+	},
+	Action: func(cli *cli.Context) error {
+		ctx := context.Background()
+		containerID := cli.Args().First()
+		if containerID == "" {
+			return errors.New("`get` command must specify a target container ID")
+		}
+		containerJobHandle, err := jobobject.GetWindowsContainerJobHandle(containerID)
```

Comment:
I think this tool would be of use for all job objects, and especially for job containers. We don't follow the same formula for naming the job so this wouldn't work for them currently

Reply:
Updated to take in a "name" instead of a containerID, where name is the target job object's name

Reply:
Thanks!